params/genesis.go

```diff
@@ -86,14 +87,19 @@ func DefaultYoloV1GenesisBlock() *genesisT.Genesis {
 
 // DeveloperGenesisBlock returns the 'geth --dev' genesis block. Note, this must
 // be seeded with the
-func DeveloperGenesisBlock(period uint64, faucet common.Address) *genesisT.Genesis {
+func DeveloperGenesisBlock(period uint64, faucet common.Address, useEthash bool) *genesisT.Genesis {
 	// Override the default period to the user requested one
-	config := *AllCliqueProtocolChanges
-	config.Clique.Period = period
+	var config *goethereum.ChainConfig
+	if useEthash {
+		config = AllEthashProtocolChanges
```

Comment:
Is the `AllEthashProtocolChanges` config going to be OK to use here? There are some considerations that Clique doesn't have to make for PoW, eg. difficulty (and difficulty bomb), block rewards.

Reply:
Good point. I haven't thought of this. I thought it will work as expected based on [initial PoW->Clique change](https://github.com/ethereum/go-ethereum/pull/15323).

Block rewards work fine. I wonder if we should care about difficulty bomb in `--dev.pow]` mode, or a good use case for it

Reply:
Dev mode shouldn't ever activate the difficulty bomb, in my opinion. And since core-geth has the ECIP available to permanently disable it, I think it should do so. We'll need to double check the `difficulty` field in the genesis block, too, to make sure it is at the `params.MinimumDifficulty` (or whatever it is (that's pseudo-code ;)). 

The thought I had in mind with the block rewards was in regards to differences available for ETH vs. ETC. For example, `AllEthashProtocolChanges` will set the reward to the ETH _Istanbul_ (permanently), while ETC might, say, set it to Era 1 of [ECIP1017](https://github.com/ethereumclassic/ECIPs/blob/master/_specs/ecip-1017.md), and would then need to establish some arbitrary value for the Era Rounds length (ie probably not 500000).

Reply:
Nice one, thanks for analysing that, it really helps me to grasp the whole system. 
I will have a look at it. It will be nice if you can give some tips on it @meowsbits , so as I learn and not spending so much time.

Reply:
Sure, check out https://github.com/etclabscore/core-geth/pull/243/commits/f4530b2c1cab406bc45f35f4429d1573490d1be5

Reply:
Thanks, very informative. EIPs used for block rewards and difficulty bomb noted. Own you a beer